### DISCUSSION TITLE
logo size in template one 3.0.5

### DISCUSSION THREAD

**Discussion by patrickus33** [templaza/astroid-framework#474] [Q&A]
HI all,
The logo size is set in the template's css.
.astroid-logo img {
  max-width: 250px;
  max-height: 65px;
}

I can't find a way to change this setting in the template interface.

Do I need to make a modification using the custom css?
It's a penalty to have "blocked" the logo size.

Thank you !

**Comment by Chacapamac** [comment]
@patrickus33 
Yes the Logo size is set in the CSS. You have to remember that the logo is comprised in the “**Section**” astroid-header-section. increasing the size (especially he height, will automatically change the size of the entire Header “**Section**”.

If You want more control over the size of the Logo you can change the positionnement of the wrapper to absolute
```
div.logo-wrapper.astroid-logo {
position: absolute;
left: 100px;
top: 50px;
}
```

and change the logo image size to your liking:
```
img.astroid-logo-default{
width: 250px;
}
```
You will have to do the same in the “Sticky” header if you want your logo the same size on both “Normal Header” & “Sticky Header”
header#astroid-sticky-header div.logo-wrapper
and
header#astroid-sticky-header img.astroid-logo-default

After you have to control all that for the different Bootstrap size and you maybe have to play with the z-index of the wrapper to keep the link on the logo working

✅ **• Enhancement**
I will mark this post as a **enhancement** for the developer { @sonvnn }
It will be great if Astroid give a possibility to control a logo size to escape the actual 250px x 65px ratio size. Not all logos (https://www.brandsoftheworld.com/) got that ratio and will simply look bad in this fixed ratio. Astroid should be able to permit an easy total control of the logo, even with a different size ratio (e.g. A round logo. Many company brand logo cannot be modified to fit that space.
_**Note:** I experiment with ALL the big Joomla Frameworks and none of them offer that possibility, they all go for that same restraint size without any solution for logos that doesn't fit that size ratio. Serious companies will not accept a website with a modified logo and demand their brand image to be as it is without compromise._

**Reply by patrickus33** [reply]
OK, thank you for so quick answer !
I'll do some CSS to fit my need .
You're right, it will be great if Astroid was the first to offer this option !

**Reply by Chacapamac** [reply]
No problem, as an example you can see this website I’m working on. (I use a different approach from above) —> http://www.graf-test.ca/

**Reply by patrickus33** [reply]
I really like this presentation with its vertical logo.
Even on a smartphone, it looks great.
It's very stylish, well done!

**Reply by Chacapamac** [reply]
Thanks @patrickus33 , Yes, quite a bit of work to make it work on all bootstrap size and in the “Sticky Header”. This is a good example of a logo that CANNOT be use with the way Astroid control the logo now. As I was saying many logos have a roundish shape and will not work. No serious company will accept any compromise with their logo.

If you have any problems don’t hesitate to come back here.

**Comment by sonvnn** [comment]
@patrickus33 @Chacapamac I added this feature in v3-branch. Please help me download v3 branch and test it on your site.
Go to Logo Options you can see these options.
![Screen Shot 2023-12-01 at 16 16 14](https://github.com/templaza/astroid-framework/assets/3204103/1d3a127e-7687-4860-a0e7-2c16c40f3904)

**Reply by patrickus33** [reply]
Hi sonvnn,

Using the v3-branch causes the crash of front with message :
ERROR 0 - Call to undefined method Joomla\Filesystem\File::getExt() in /home/zdnbqnqc/v2/libraries/astroid/framework/frontend/logo.php:118
